pytorch_lightning/plugins/pipe_plugin.py

```diff
@@ -0,0 +1,356 @@
+        return self._use_barrier_and_broadcast
+
+    @property
+    def use_optimizer_step(self):
+        return self.version != 1
+
+    def _infering_balance_from_example_input_array(self, trainer):
+        model = trainer.get_model()
+        if not hasattr(model, "layers") or not isinstance(model.layers, nn.Sequential):
+            raise MisconfigurationException(
+                'Could not find a PipeLightningModule within the model. '
+                'Did you defined set your sequential model as an `layers` attribute of your model ?')
+
+        if self.balance is None:
+            partitions = torch.cuda.device_count() if self.num_partitions is None else self.num_partitions
+            if model.example_input_array is not None:
+                balance_func = getattr(pipe_balance, self.balance_mode)
+                self.balance = balance_func(partitions, model.layers, model.example_input_array)
+                log.info(f'The following balance {self.balance} was inferended using {self.balance_mode} mode')
+            else:
+                raise MisconfigurationException(
+                    'Please, set example_input_array to your model, so we can infer the right balance for you')
+
+    def _find_pipe_module(self, model):
+        # try to wrap for the user
+        if hasattr(model, "layers") and isinstance(model.layers, nn.Sequential):
+            model.layers = LightningPipeModule(
+                model.layers,
+                balance=self.balance,
+                microbatches=self.microbatches,
+                checkpoint=self.checkpoint,
+                version=self.version,
+            )
+            model.final_stage = model.layers.module.final_stage
+            if self.version == 1:
+                model.back_helper = model.layers.module.back_helper
+            else:
+                model.foreach_worker = model.layers.module.foreach_worker
+                model.layers.module.model.trainer = model.trainer
+                model.layers.module.model.configure_optimizers = model.configure_optimizers
+            pipe_module = model
+            found_module = True
+
+        if not found_module:
+            raise MisconfigurationException(
+                'Could not find a PipeLightningModule within the model. '
+                'Did you defined set your sequential model as an `layers` attribute of your model ?')
+        return pipe_module
+
+    def init_ddp_connection(
+            self,
+            trainer,
+            cluster_environment,
+            global_rank: int,
+            world_size: int,
+            is_slurm_managing_tasks: bool = True,
+    ) -> None:
+        super().init_ddp_connection(
+            trainer=trainer,
+            cluster_environment=cluster_environment,
+            global_rank=global_rank,
+            world_size=world_size,
+            is_slurm_managing_tasks=is_slurm_managing_tasks
+        )
+        os.environ["MASTER_PORT"] = "15000"
```

Comment:
Eventually we'll need a strategy for this

Reply:
I tried to use random port, but not really reliable I think